### DISCUSSION TITLE
[General] query the status of transactions or blocks

### DISCUSSION THREAD

**Discussion by Poetry-Distance** [zkSync-Community-Hub/zksync-developers#698] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

Which API should be used to query the status of transactions or blocks

**Comment by alhenaai** [comment]
To query the status of transactions or blocks on ZKsync, you can use the ZKsync Era Block Explorer API. This API allows developers to access comprehensive data about transactions, blocks, batches, wallets, tokens, and smart contracts on the ZKsync Era network. You can find more details and how to use the API in the [Block Explorer API documentation](https://docs.zksync.io/build/tooling/zksync-block-explorers#block-explorer-api).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
You can use use the regular [eth_getTransactionReceipt](https://docs.zksync.io/build/api-reference/ethereum-rpc#eth_gettransactionreceipt) for transactions and [eth_getBlockByNumber](https://docs.zksync.io/build/api-reference/ethereum-rpc#eth_getblockbynumber).

Also the ZKsync-specific ones, [`zks_getTransactionDetails`](https://docs.zksync.io/build/api-reference/zks-rpc#zks_gettransactiondetails) and [`zks_getBlockDetails`](https://docs.zksync.io/build/api-reference/zks-rpc#zks_getblockdetails)
